src/benchmarking/randomized_benchmarking.jl

```diff
@@ -0,0 +1,342 @@
+
+        for j_length = 2:properties.sequence_length_list[sequence_length_id]
+            new_clifford = get_random_clifford(num_target_qubits)
+            new_circuit = get_transpiled_circuit(new_clifford, properties, qubit_map,
+                transpile!)
+            append!(circuit, new_circuit)
+            clifford_product = clifford_product*new_clifford
+            old_clifford = new_clifford
+        end
+        inverse_clifford = inv(clifford_product)
+        inverse_circuit = get_transpiled_circuit(inverse_clifford, properties,
+            qubit_map, transpile!)
+        append!(circuit, inverse_circuit)
+        push!(circuit_list, circuit)
+    end  
+    return circuit_list
+end
+
+function get_transpiled_circuit(clifford, properties, qubit_map, transpile!)
+    num_target_qubits = length(qubit_map)
+    clifford_circuit = QuantumCircuit(qubit_count=num_target_qubits,
+            bit_count=properties.num_bits_on_device)
+    push_clifford!(clifford_circuit, clifford)
+    reordered_circuit = get_reordered_circuit(clifford_circuit, qubit_map)
+    circuit = get_wider_circuit(reordered_circuit, properties.num_qubits_on_device)
+    transpile!(circuit)
+end
+
+function get_fitting_model_results(sequence_fidelities,
+    properties::RandomizedBenchmarkingProperties,
+    fit_properties::RandomizedBenchmarkingFitProperties)
+
+    order = fit_properties.model_order
+    results = RandomizedBenchmarkingFitResults()
+    if order == 0
+        initial_parameters = [fit_properties.initial_parameters["p"],
+            fit_properties.initial_parameters["A0"],
+            fit_properties.initial_parameters["B0"]]
+        fit = curve_fit(get_zeroth_model, properties.sequence_length_list,
+            sequence_fidelities, initial_parameters)
+        optimal_parameters = Dict("p"=>fit.param[1], "A0"=>fit.param[2], "B0"=>fit.param[3])
+        results = RandomizedBenchmarkingFitResults(model_order=order,
+            parameters=optimal_parameters,
+            residuals=fit.resid, jacobian=fit.jacobian,
+            converged=fit.converged)
+    elseif order == 1
+        initial_parameters = [fit_properties.initial_parameters["p"],
+            fit_properties.initial_parameters["q"],
+            fit_properties.initial_parameters["A1"],
+            fit_properties.initial_parameters["B1"],
+            fit_properties.initial_parameters["C1"]]
+        fit = curve_fit(get_first_model, properties.sequence_length_list,
+            sequence_fidelities, initial_parameters)
+        LsqFit.LsqFitResult
+        optimal_parameters = Dict("p"=>fit.param[1], "q"=>fit.param[2], "A1"=>fit.param[3],
+            "B1"=>fit.param[4], "C1"=>fit.param[5])
+        results = RandomizedBenchmarkingFitResults(model_order=order,
+            parameters=optimal_parameters,
+            residuals=fit.resid, jacobian=fit.jacobian,
+            converged=fit.converged)
+    end
+    return results
+end
+
+function get_zeroth_model(m, p)
```

Comment:
What is the data type of m and p? There is absolutely no documentation here. Moreover, I noticed unit tests on my Mac fail on this function due to Segmentation Fault.

Reply:
This function is not meant to be directly called by a user, which is why I haven't added documentation for it. The documentation for RandomizedBenchmarkingFitResults gives a description of each model including this one. I will add type declarations for m and p.